compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -907,4 +907,28 @@ object JavaParsers {
       unit
     }
   }
+
+
+  /**
+   * OutlineJavaParser parses top-level declarations in `source` to find declared classes, ignoring their bodies (which
+   * must only have balanced braces). This is used to map class names to defining sources.
+   * This is necessary even for Java, because the filename defining a non-public classes cannot be determined from the
+   * classname alone.
+   */
+  class OutlineJavaParser(source: SourceFile)(implicit ctx: Context) extends JavaParser(source) {
+
+    def skipBraces[T](body: T): T = {
```

Comment:
I realise it was taken from the Scala `OutlineParser `, but I find it a bit weird that this method takes an argument and just returns it. I would define it and use as:
```scala
private def skipBraces(): Unit = ...
override def typeBody(leadingToken: Int, parentName: Name, parentTParams: List[TypeDef]) = {
  skipBraces()
  (List(EmptyValDef), List(EmptyTree))
}
  
```
Maybe change the implementation in the Scala OutlineParser as well.